api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -98,22 +98,36 @@ public void bufferBillingProjects() {
         billingProjectBufferEntryDao.getBillingBufferGaugeData().stream()
             .collect(
                 Collectors.toMap(
-                    c -> Pair.of(c.getAccessTier(), c.getStatusEnum()),
+                    c -> {
+                      String logMsg =
+                          String.format(
+                              "counts Tier %s / Status %s: Count %d",
+                              c.getAccessTier().getShortName(),
+                              c.getStatusEnum().toString(),
+                              c.getNumProjects());
+                      log.info(logMsg);
+
+                      return Pair.of(c.getAccessTier(), c.getStatusEnum());
+                    },
                     ProjectCountByStatusAndTier::getNumProjects));
 
     // iterate through ALL combinations of access tier and buffer status, counting 0 projects where
     // this combination does not appear in `counts`
-    return Streams.zip(
-            accessTierService.getAllTiers().stream(),
-            Arrays.stream(BufferEntryStatus.values()),
-            (tier, status) -> {
-              final long numProjects = counts.getOrDefault(Pair.of(tier, status), 0L);
-              return MeasurementBundle.builder()
-                  .addMeasurement(GaugeMetric.BILLING_BUFFER_PROJECT_COUNT, numProjects)
-                  .addTag(MetricLabel.BUFFER_ENTRY_STATUS, status.toString())
-                  .addTag(MetricLabel.ACCESS_TIER_SHORT_NAME, tier.getShortName())
-                  .build();
-            })
+    return accessTierService.getAllTiers().stream()
+        .flatMap(
```

Comment:
this is how you actually combine two streams multiplicatively.

alt: don't use streams

Reply:
nooo streams are great